source/_posts/2019-02-04-introducing-home-assistant-cli.markdown

```diff
@@ -0,0 +1,180 @@
+date: 2019-02-04 00:01:00
+date_formatted: "February 4, 2018"
+author: Max Rydahl Andersen
+author_twitter: maxandersen
+comments: true
+categories: Announcements
+---
+
+When I started using Home Assistant (HA) I found the browser UI great for casual
+usecases. For my usecases it felt a bit slow when trying to debug/trace what
+was going on. I realized I was missing a way to interact with HA via a command
+line (CLI) with the power of auto-completion and direct access without having to wait
+for pages to load and clicking around.
+
+Thus I started working on [Home Assistant CLI][github-hass-cli], a project that
+was initially started by [Fabian Affolter][@fabaff].
+
+Last week we released version 0.4, which provides majority of the features I was
+looking for to do from a CLI.
+
+## Feature highlights:
+   
+   - Get configuration info (`hass-cli config`)
+   - List, get, edit and delete state for entities (`hass-cli entity`')
+   - Query history with relative time ('hass-cli entity history')
+   - List and run services (`hass-cli services`)
+   - Execute templates locally and remotely (`hass-cli template`)
+   - Control over columns, sorting, etc. (`hass-cli --columns attr1,attr2 --sort-by attr3`)
+   - Shell completion for most commands and arguments (`hass-cli completion zsh`)
+   - Get logs (`hass-cli system log`)
+   - Run discovery (`hass-cli discover`)
+   - Show map for zone (`hass-cli map`)
+   - Call raw api directly (`hass-cli raw`)
+   - ... and more
+
+## Usage
+
+For the basic intro to `hass-cli` see the docs at https://github.com/home-assistant/home-assistant-cli.
+
+## Installation
+
+Install latest `homeassistant-cli` by pip:
+
+    $ pip3 install homeassistant-cli
+   
+or if you like an isolated install you can use [pipsi][pipsi]
+
+    $ pipsi install --python python3 homeassistant-cli
+
+or if you are adventurous you can build it from [source][github-hass-cli] at. 
+
+## Changes since 0.1-0.3
+
+The first thing is that the commands have been cleaned up to be more logcial/explicit. 
+
+    $ hass-cli 
+    ...
+    ...
+    Commands:
+    completion  Output shell completion code for the specified shell (bash or...
+    config      Get configuration from a Home Assistant instance.
+    discover    Discovery for the local network.
+    entity      Get info and operate on entities from Home Assistant.
+    event       Interact with events.
+    info        Get basic info from Home Assistant.
+    map         Print the current location on a map.
+    raw         Call the raw API (advanced).
+    service     Call and work with services.
+    system      System details and operations for Home Assistant.
+    template    Render templates on server or locally.
+
+Rather than using generic terms like get, list, etc. The commands are now explicitly
+named what they are about, and then have further sub-command.
+
+The biggest behavior change is that most operations by default now uses
```

Comment:
The biggest change is that most operations, by default, now use "table" mode, i.e., instead of getting a yaml or json dump you get a more concise view:

Reply:
fixed.